src/cpp/r/session/RStdCallbacks.cpp

```diff
@@ -426,8 +426,20 @@ void RWriteConsoleEx (const char *buf, int buflen, int otype)
    {
       if (!s_suppressOutput)
       {
+         bool isInterruptOutput =
```

Comment:
This is sort of a hack based on how we draw the prompt. In a regular R session, when an interrupt is issued, R writes a single newline:

https://github.com/wch/r-source/blob/819477cd6dbefb8fae61a07fc13064faa29a2714/src/main/errors.c#L188

with the intention of having that ensure the input prompt is then drawn on its own line. The net result is that interrupting a command gives you a "new" prompt to work on.

RStudio instead tries a little harder to ensure the prompt is always shown on its own line, which means an extraneous newline is written to the console history in these cases. Hence, this is ultimately a hack to ensure that RStudio's output more closely mimics that of R.